backend/api/documents/v3alpha/resources.go

```diff
@@ -2,23 +2,137 @@ package documents
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"io"
 	"net/url"
 	"seed/backend/blob"
 	"seed/backend/core"
 	documents "seed/backend/genproto/documents/v3alpha"
+	p2p "seed/backend/genproto/p2p/v1alpha"
+	"seed/backend/hmnet/syncing"
 	"seed/backend/util/dqb"
 	"seed/backend/util/sqlite"
 	"seed/backend/util/sqlite/sqlitex"
 	"slices"
+	"time"
 
 	"github.com/fxamacker/cbor/v2"
 	"github.com/ipfs/go-cid"
 	cbornode "github.com/ipfs/go-ipld-cbor"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
+// PushResourcesToPeer implements the corresponding gRPC method.
+func (srv *Server) PushResourcesToPeer(req *documents.PushResourcesToPeerRequest, streamTx grpc.ServerStreamingServer[documents.SyncingProgress]) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
```

Comment:
Do we need this strict timeout here? I'm afraid sometimes 30 seconds may be too low. What if the connection is slow, or there's a lot of data to sync?